src/views/ScreeningsPage.vue

```diff
@@ -1,13 +1,83 @@
-<script>
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { computed, ref, watchEffect } from 'vue';
 
-export default defineComponent({
-  name: 'ScreeningsPage'
+import { dayNameENG } from '@/helpers/date';
+
+import BaseHeader from '@/components/global/BaseHeader.vue';
+import ScreeningsSearchBar from '@/components/Screenings/ScreeningsSearchBar.vue';
+import ScreeningsCard from '@/components/Screenings/ScreeningsCard.vue';
+
+import { useScreeningsStore } from '@/store/screenings';
+import { useMoviesStore } from '@/store/movies';
+
+import type { Screening } from '@/types/screening';
+
+const screenings = useScreeningsStore();
+const movies = useMoviesStore();
+
+const selectedMovie = ref('All movies');
+
+const headerDate = computed(() => {
+  const date = new Date(screenings.currentDate);
+  const weekday = dayNameENG(date, 'long');
+  const year = date.getFullYear();
+  const month = String(date.getMonth() + 1).padStart(2, '0');
+  const day = String(date.getDate()).padStart(2, '0');
+  const capitalizedDay = weekday.charAt(0).toUpperCase() + weekday.slice(1);
+  return `${capitalizedDay} ${day}/${month}/${year}`;
+});
+
+const moviesByTitle = computed(() => {
+  return selectedMovie.value === 'All movies'
+    ? movies.allMovies
+    : movies.allMovies.filter((movie) => movie.title === selectedMovie.value);
+});
+
+const displayMovies = computed(() => {
+  const movies = screenings.currentScreenings.map(
+    (screening) => screening.movie
+  );
+  return moviesByTitle.value.filter((movie) => movies.includes(movie.id));
+});
+
+function getMovieScreenings(movieId: number): Screening[] {
+  return screenings.currentScreenings.filter(
+    (screening) => screening.movie === movieId
+  );
+}
+
+watchEffect(async () => {
```

Comment:
If you don't use `await` then it's not needed 😄 
```suggestion
watchEffect(() => {
```

Reply:
Ok, just so I get this once in for all - when I declare an async function in a store / api file then I don't need to call it asynchronously in the component itself?

Reply:
It's up to you - the async function will be returning a promise but we can decide if we just want to execute it or await the result; it all depends on what we want to achieve 😄

Reply:
Q answered on summary --> resolving 😃